test/sdk-benchmarks/src/main/java/software/amazon/awssdk/benchmark/utils/StreamingMockServlet.java

```diff
@@ -0,0 +1,112 @@
+import javax.servlet.http.HttpServletResponse;
+
+public class StreamingMockServlet extends HttpServlet {
+    private static final byte[] STREAMING_RESPONSE_DATA = new byte[1024 * 1024]; // 1MB response
+
+    static {
+        // Initialize response data
+        for (int i = 0; i < STREAMING_RESPONSE_DATA.length; i++) {
+            STREAMING_RESPONSE_DATA[i] = (byte) (i % 256);
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        handleRequest(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        handleRequest(request, response);
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws  IOException {
+        handleRequest(request, response);
+    }
+
+    private void handleRequest(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Check if this should be a streaming response
+        if (isStreamingOperation(request)) {
+            handleStreamingRequest(request, response);
+        } else {
+            handleJsonRequest(request, response);
+        }
+    }
+
+    private boolean isStreamingOperation(HttpServletRequest request) {
+        String uri = request.getRequestURI();
+        String contentType = request.getContentType();
+
+        return uri.contains("streaming") ||
+               uri.contains("StreamingInput") ||
+               uri.contains("StreamingOutput") ||
+               "application/octet-stream".equals(contentType);
+    }
+
+    private void handleStreamingRequest(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Consume input stream if present
+        try (InputStream inputStream = request.getInputStream()) {
+            byte[] buffer = new byte[8192];
+            while (inputStream.read(buffer) != -1) {
+                // Just consume the data
+            }
+        }
+
+        // Send streaming response
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.setContentType("application/octet-stream");
+        response.setContentLength(STREAMING_RESPONSE_DATA.length);
+        response.setHeader("x-amz-request-id", "streaming-" + System.currentTimeMillis());
+
+        try (OutputStream outputStream = response.getOutputStream()) {
+            outputStream.write(STREAMING_RESPONSE_DATA);
+            outputStream.flush();
+        }
+    }
+
+    private void handleJsonRequest(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.setContentType("application/json");
+        response.setCharacterEncoding("UTF-8");
+        response.setHeader("x-amz-request-id", "json-" + System.currentTimeMillis());
+
+        String jsonResponse = "{"
```

Comment:
Dumy responses this will make sure the Client will call the ResponseBody.getContent and test Nenchmarks how  the client buffers